packages/minifier-css/minifier-async-tests.js

```diff
@@ -0,0 +1,51 @@
+import { CssTools } from './minifier';
+const TEST_CASES = [
+  ['a \t\n{ color: red } \n', 'a{color:red}', 'whitespace check'],
+  [
+    'a \t\n{ color: red; margin: 1; } \n',
+    'a{color:red;margin:1}',
+    'only last one loses semicolon',
+  ],
+  [
+    'a \t\n{ color: red;;; margin: 1;;; } \n',
+    'a{color:red;margin:1}',
+    'more semicolons than needed',
+  ],
+  ['a , p \t\n{ color: red; } \n', 'a,p{color:red}', 'multiple selectors'],
+  ['body {}', '', 'removing empty rules'],
+  [
+    '*.my-class { color: #fff; }',
+    '.my-class{color:#fff}',
+    'removing universal selector',
+  ],
+  [
+    'p > *.my-class { color: #fff; }',
+    'p>.my-class{color:#fff}',
+    'removing optional whitespace around ">" in selector',
+  ],
+  [
+    'p +  *.my-class { color: #fff; }',
+    'p+.my-class{color:#fff}',
+    'removing optional whitespace around "+" in selector',
+  ],
+  [
+    'a {\n\
+  font:12px \'Helvetica\',"Arial",\'Nautica\';\n\
+  background:url("/some/nice/picture.png");\n}',
+    'a{font:12px Helvetica,Arial,Nautica;background:url(/some/nice/picture.png)}',
+    'removing quotes in font and url (if possible)',
+  ],
+  ['/* no comments */ a { color: red; }', 'a{color:red}', 'remove comments'],
+];
+
+Tinytest.addAsync(
+  '[Async] minifier-css - simple CSS minification',
+  (test, onComplete) => {
+    const promises = TEST_CASES.map(([css, expected, desc]) =>
+      CssTools.minifyCssAsync(css).then((minifiedCss) => {
+        test.equal(minifiedCss[0], expected, desc);
+      })
+    );
+    Promise.all(promises).then(() => onComplete());
```

Comment:
`addAsync` also works with promises, therefore this test could look like this:
```suggestion
  async (test) => {
    for (const [css, expected, desc] of TEST_CASES) {
      const minifiedCss = await CssTools.minifyCssAsync(css);
      test.equal(minifiedCss[0], expected, desc);
    }
```

It also fixes the problem that right now it doesn't handle rejection correctly.

Reply:
I changed the code to handle rejections but didn't use await because I wanted execution out of order.